src/aave-v2/lens/RatesLens.sol

```diff
@@ -359,9 +424,51 @@ abstract contract RatesLens is UsersLens {
         );
     }
 
-    /// INTERNAL ///
+    /// @dev Computes and returns the underlying pool rates for a specific market.
+    /// @param _underlying The underlying pool market address.
+    /// @param _supplied The amount hypothetically supplied (in underlying).
+    /// @param _borrowed The amount hypothetically borrowed (in underlying).
+    /// @param _repaid The amount hypothetically repaid (in underlying).
+    /// @param _withdrawn The amount hypothetically withdrawn (in underlying).
+    /// @return poolSupplyRate The market's pool supply rate per year (in ray).
+    /// @return poolBorrowRate The market's pool borrow rate per year (in ray).
+    function _getPoolRatesPerYear(
+        address _underlying,
+        uint256 _supplied,
+        uint256 _borrowed,
+        uint256 _withdrawn,
+        uint256 _repaid
+    ) internal view returns (uint256 poolSupplyRate, uint256 poolBorrowRate) {
+        DataTypes.ReserveData memory reserve = pool.getReserveData(_underlying);
+
+        PoolRatesVars memory vars;
+        (
+            vars.availableLiquidity,
+            vars.totalStableDebt,
+            vars.totalVariableDebt,
+            ,
+            ,
+            ,
+            vars.avgStableRate,
+            ,
+            ,
+
+        ) = dataProvider.getReserveData(_underlying);
```

Comment:
I believe that these variables can be fetched trivially without the need to use the data provider. The data provider does not seem to be a contract meant to interface with other contracts since many of the calls are very heavy and fetch a lot of extra data.

Reply:
Indeed, it could be replaced with 3 calls, according to AaveV2's source code:
- https://github.com/aave/protocol-v2/blob/master/contracts/protocol/libraries/logic/ReserveLogic.sol#L209-L217
- https://github.com/aave/protocol-v2/blob/ce53c4a8c8620125063168620eba0a8a92854eb8/contracts/protocol/lendingpool/DefaultReserveInterestRateStrategy.sol#L130

And at the same time, would avoid relying on the data provider / having its address as immutable

Is it worth it though?

Reply:
After having dived into the traces, the data provider is just a proxy for these 3 calls. Applying your suggestion and getting rid of the data provider